FindMKL.cmake

```diff
@@ -0,0 +1,222 @@
+#[=======================================================================[.rst:
+FindMKL
+-------
+
+Example usage
+^^^^^^^^^^^^^
+
+  find_package(MKL REQUIRED OPTIONAL_COMPONENTS SCALAPACK)
+  target_link_libraries(... MKL::MKL)
+  target_link_libraries(... MKL::SCALAPACK)
+
+Note: Targets linking to MKL::SCALAPACK need not link to MKL::MKL. It is NOT 
+      an error if both targets are linked though.
+
+
+Options
+^^^^^^^
+
+``MKLROOT``
+  Environment variable set to MKL's root directory  
+
+``MKL_ROOT``
+  CMake variable set to MKL's root directory  
+
+``MKL_PARALLEL``           
+  ON|OFF   (default: ON / parallel)
+
+``MKL_64BIT``
+  ON|OFF   (default: OFF / 32bit interface)
+
+
+Components
+^^^^^^^^^^
+
+``SCALAPACK``
+
+
+Imported targets 
+^^^^^^^^^^^^^^^^
+
+``MKL::MKL``
+
+``MKL::SCALAPACK``
+
+
+Result variables
+^^^^^^^^^^^^^^^^
+
+``MKL_FOUND``
+  Found MKL.
+
+``MKL_SCALAPACK_FOUND``
+  Found ScaLAPACK.
+
+Not supported
+^^^^^^^^^^^^^
+
+- TBB threading back-end
+- F95 interfaces
+
+Note: Mixing GCC and Intel OpenMP backends is a bad idea.
+
+#]=======================================================================]
+
+# Modules
+#
+include(FindPackageHandleStandardArgs)
+
+# Functions
+#
+function(__mkl_find_library _name)
+    find_library(${_name}
+        NAMES ${ARGN}
+        HINTS ${MKL_ROOT}
+        PATH_SUFFIXES ${_mkl_libpath_suffix}
+                      lib
+        )
+    mark_as_advanced(${_name})
+endfunction()
+
+# Options
+#
+# The `NOT DEFINED` guards on CACHED variables are needed to make sure that 
+# normal variables with of same name always take precedence*.
+#
+# * There are many caveats with CACHE variables in CMake. Before version 
+#   3.12, both the `option()` command and `set(... CACHE ...)` would 
+#   override normal variables if cached equivalents don't exist or they 
+#   exisit but their type is not specified (e.g. command line arguments: 
+#   -DFOO=ON instead of -DFOO:BOOL=ON). For 3.13 with policy CMP0077, 
+#   `option()` no longer overrides normal variables of the same name.
+#   `set(... CACHE ...)` is still stuck with the old behaviour. 
+#
+#   https://cmake.org/cmake/help/v3.15/command/set.html#set-cache-entry
+#   https://cmake.org/cmake/help/v3.15/policy/CMP0077.html
+#
+if(NOT DEFINED MKL_ROOT)
+    set(MKL_ROOT $ENV{MKLROOT} CACHE PATH "MKL's root directory.")
+endif()
+
+if(NOT DEFINED MKL_PARALLEL)
+    set(MKL_PARALLEL ON CACHE BOOL "Toggles parallel and sequential versions.")
+endif()
+
+if(NOT DEFINED MKL_64BIT)
+    set(MKL_64BIT OFF CACHE BOOL "Toggles 32bit/64bit MKL interfaces.")
+endif()
+
+# Determine MKL's library folder
+#
+set(_mkl_libpath_suffix "lib/intel64")
+if(CMAKE_SIZEOF_VOID_P EQUAL 4) # 32 bit
```

Comment:
You are not very consistent with spacings. I prefer having no spaces between command name and parentheses